plugins/dereferenceUses.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const { querySelectorAll, querySelector } = require('../lib/xast.js');
+
+exports.type = 'full';
+
+exports.active = true;
+
+exports.description = 'dereferences <use/> elements';
+
+exports.params = {
+  keepHref: false, // keep (xlink:)href attributes
+
+  symbolContainer: 'svg', // browsers use <svg/> as container of <symbol/> content (`g` could also be used)
+};
+
+const OverridingUseAttributeNames = [
+  'x',
+  'y',
+  'width',
+  'height',
+  'href',
+  'xlink:href',
+];
+const HrefAttributeNames = ['href', 'xlink:href'];
+
+/**
+ * Dereferences <use> elements
+ *
+ *
+ * @param {Object} document document element
+ * @param {Object} options plugin params
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document, options) {
+  options = options || {};
+
+  // collect <use/>s
+  const useElements = querySelectorAll(document, 'use');
+
+  // no <use/>s, nothing to do
+  if (useElements === null) {
+    return document;
+  }
+
+  // replace each <use/> with its referenced node
+  for (const useElement of useElements) {
+    // `href`/`xlink:href` value
+    const hrefAttribute = useElement.attr('href')
+      ? useElement.attr('href')
+      : useElement.attr('xlink:href');
```

Comment:
Access attributes directly

Reply:
OK. Also the code for finding a valid href attribute was improved.